pkg/provisioner/crdprovisioner.go

```diff
@@ -154,40 +155,29 @@ func (c *CrdProvisioner) CreateVolume(
 	volumeName = azureutils.CreateValidDiskName(volumeName, true)
 	azVolumeName := strings.ToLower(volumeName)
 
-	waiter, err := c.conditionWatcher.newConditionWaiter(ctx, azVolumeType, azVolumeName, func(obj interface{}, _ bool) (bool, error) {
-		if obj == nil {
-			return false, nil
-		}
-		azVolumeInstance := obj.(*diskv1beta1.AzVolume)
-		if azVolumeInstance.Status.Detail != nil {
-			return true, nil
-		} else if azVolumeInstance.Status.Error != nil {
-			return false, util.ErrorFromAzError(azVolumeInstance.Status.Error)
-		}
-		return false, nil
-	})
-	if err != nil {
-		return nil, err
-	}
-	defer waiter.Close()
-
-	azVolumeInstance, err := lister.Get(azVolumeName)
+	ctx, workflow := azureutils.StartWorkflow(ctx, "CrdProvisioner.CreateVolume", consts.VolumeNameLabel, volumeName, consts.PvNameKey, parameters[consts.PvNameKey])
```

Comment:
Rather than require the caller to pass in the function name, we could use [`runtime.Caller(1)`](https://pkg.go.dev/runtime#Caller) and [`runtime.FuncForPC()`](https://pkg.go.dev/runtime#FuncForPC) to discover the calling function name. We would end up with a name like `provisioner.(*CrdProvisioner).CreateVolume`.